### DISCUSSION TITLE
What is there and who owns it? Should there be anything else?

### DISCUSSION THREAD

**Discussion by sndrs** [guardian/csnx#430] [Team Posts]
- should we establish a new role of maintainer(s), in an OSS sense? 
  - cross-team members would maintain projects
- what do we do with old projects (and ones that become old)?
- what things are _already_ out there? how are they discoverable?
- how do we drive adoption of common libs?

**Comment by sndrs** [comment]
i wonder if just using github codeowners would be sufficient for handling who's adopted which projects? 

- only codeowners can merge
- anyone can PR
- anyone can PR themselves to become a codeowner or stop being one

idea being that it would:

- make it clear which people are 'looking after' something, esp if you're chasing a change
- no one has to be owner of anything they don't want to be
- becomes clear if a project needs more people/time
- everyone can contribute, but the project has a long-view eyes on it as well as on individual changes

taking https://github.com/guardian/fontastique/pull/28 as an example: 

- i'd raise a PR either after a chat or as a suggestion
- if @JamieB-gu was happy with it, he'd just merge it, or request changes etc

i have the immediate problem that needs solving and i fix it, but @JamieB-gu understands it in the larger context. jamie's not expected to make all the changes though.

any thoughts?

**Comment by JamieB-gu** [comment]
Seems sensible to me. I wonder if we could use the GH API to create a dashboard of code-owners for a project, so anyone looking for help can easily find the right person.

> only codeowners can merge

Do we need this restriction for this sort of project? 🤔 I guess it depends on how much damage an accidental merge could cause.

**Comment by sndrs** [comment]
i feel like it would be prudent. it means the owners know what is going in, and they're responsible for managing releases too. if anyone was able to merge i can kind of see the long-term view they have being lost?

**Comment by sndrs** [comment]
although maybe something like [semantic release](https://github.com/semantic-release/semantic-release) and  requiring reviews would accomplish the same thing while being less restricitive?

**Comment by JamieB-gu** [comment]
If we can require review from a given list of maintainers/owners, perhaps that will be enough to give us the oversight we need?

GitHub already has the ability to distinguish between different reviewers AFAIK - I think maintainers get a green tick and non-maintainers get a grey tick?
